skimage/morphology/footprints.py

```diff
@@ -251,13 +523,15 @@ def ball(radius, dtype=np.uint8):
     return np.array(s <= radius * radius, dtype=dtype)
 
 
-def octagon(m, n, dtype=np.uint8):
+def octagon(m, n, dtype=np.uint8, *, decomposition=None):
     """Generates an octagon shaped footprint.
 
     For a given size of (m) horizontal and vertical sides
     and a given (n) height or width of slanted sides octagon is generated.
     The slanted sides are 45 or 135 degrees to the horizontal axis
-    and hence the widths and heights are equal.
+    and hence the widths and heights are equal. The overall size of the
+    footprint will be ``m + 2 * n`` (note that `m` must be odd to get an
+    odd-sized footprint).
```

Comment:
It is a little bit confusing: this can be interpreted as `m must be odd` but I think that you mean `If an odd output size is needed, m must be odd`, right?

Reply:
Yes. perhaps we don't need to explicitly state that and can just leave it at:

```
The overall size of the footprint along a single axis will be ``m + 2 * n``.
```

Reply:
:+1: Yes, this is in fact much more explicit :wink: